build_tools/aws-sdk-code-generator/lib/aws-sdk-code-generator/views/endpoints_module.rb

```diff
@@ -34,157 +34,35 @@ def module_name
-        ] unless value
-
-        [value || 'nil', source]
-      end
-
-      def client_context_param_value(param_name, param_data)
-        if @service.api['clientContextParams']&.key?(param_name) &&
-           !param_data['builtIn']
-          "context.config.#{Underscore.underscore(param_name)}"
-        end
-      end
-
-      def built_in_client_context_param_value(param_data)
-        case param_data['builtIn']
-        when 'AWS::Region'
-          'context.config.region'
-        when 'AWS::UseFIPS'
-          'context.config.use_fips_endpoint'
-        when 'AWS::UseDualStack'
-          if @service.name == 'S3' || @service.name == 'S3Control'
-            'context[:use_dualstack_endpoint]'
-          else
-            'context.config.use_dualstack_endpoint'
-          end
-        when 'AWS::Auth::AccountId'
-          'context.config.credentials.credentials.account_id'
-        when 'AWS::Auth::AccountIdEndpointMode'
-          'context.config.account_id_endpoint_mode'
-        when 'AWS::STS::UseGlobalEndpoint'
-          "context.config.sts_regional_endpoints == 'legacy'"
-        when 'AWS::S3::UseGlobalEndpoint'
-          "context.config.s3_us_east_1_regional_endpoint == 'legacy'"
-        when 'AWS::S3::Accelerate'
-          if @service.name == 'S3' || @service.name == 'S3Control'
-            'context[:use_accelerate_endpoint]'
-          else
-            'context.config.use_accelerate_endpoint'
-          end
-        when 'AWS::S3::ForcePathStyle'
-          'context.config.force_path_style'
-        when 'AWS::S3::UseArnRegion', 'AWS::S3Control::UseArnRegion'
-          'context.config.s3_use_arn_region'
-        when 'AWS::S3::DisableMultiRegionAccessPoints'
-          'context.config.s3_disable_multiregion_access_points'
-        when 'SDK::Endpoint'
-          'context.config.regional_endpoint ? nil : context.config.endpoint.to_s'
-        end
-      end
-
-      def context_param_value(operation, param_name)
-        return nil unless operation['input']
-
-        input_shape = operation['input']['shape']
-        members = @service.api['shapes'][input_shape].fetch('members', {})
-        members.detect do |(member_name, member)|
-          context_param = member.fetch('contextParam', {})
-          if context_param.fetch('name', nil) == param_name
-            break "context.params[:#{Underscore.underscore(member_name)}]"
-          end
-        end
-      end
-
-      def operation_context_param_value(operation, param_name)
-        return nil unless operation['input']
-
-        binding = operation.fetch('operationContextParams', {})[param_name]
-
-        return nil unless binding
-
-        "JMESPath.search(\"#{Underscore.underscore_jmespath(binding['path'])}\", context.params)"
-      end
-
-      def static_context_param(operation, param_name)
-        operation.fetch('staticContextParams', {})
-                 .fetch(param_name, {}).fetch('value', nil)
+        end.select { |p| p.source == 'operation' }
```

Comment:
This could probably be moved to the loop for readability and efficiency

Reply:
how? I guess I could return a nil and then compact instead of select? But that doesn't seem any better.

Reply:
It would have to go back to each with object instead of map, and then next item if source is not operation. If it's operation then add the parameter.

Reply:
Hmm, that seems a lot more complicated than using a simple map/select, but happy to change it back if you have strong feelings.

Reply:
Changed to each.with_object.